notebooks/working/investigating_deconfounder.py

```diff
@@ -638,39 +684,28 @@ def interceptor(rv_constructor, *rv_args, **rv_kwargs):
 
 
 # +
-def ppca_model(data_dim, latent_dim, num_datapoints, stddv_datapoints, holdout_mask):
+def ppca_model(data_dim, latent_dim, num_datapoints, stddv_datapoints, holdout_mask, linear = True):
     w = ed.Normal(loc=tf.zeros([latent_dim, data_dim]),
                 scale=tf.ones([latent_dim, data_dim]),
                 name="w")  # parameter
     z = ed.Normal(loc=tf.zeros([num_datapoints, latent_dim]),
                 scale=tf.ones([num_datapoints, latent_dim]), 
                 name="z")  # local latent variable / substitute confounder
-    x = ed.Normal(loc=tf.multiply(tf.matmul(z, w), 1-holdout_mask),
+    if linear:
+        x = ed.Normal(loc=tf.multiply(tf.matmul(z, w), 1-holdout_mask),
                 scale=stddv_datapoints * tf.ones([num_datapoints, data_dim]),
                 name="x")  # (modeled) data
-    return x, (w, z)
-
-
+    else: 
+        x = ed.Normal(loc=tf.multiply(tf.matmul(z, w), 1-holdout_mask) 
+                      + tf.multiply(tf.matmul(z**2, w**2), 1-holdout_mask),
+                scale=stddv_datapoints * tf.ones([num_datapoints, data_dim]),
+                name="x")
 
-def variational_model(qw_mean, qw_stddv, qz_mean, qz_stddv):
-    qw = ed.Normal(loc=qw_mean, scale=qw_stddv, name="qw")
-    qz = ed.Normal(loc=qz_mean, scale=qz_stddv, name="qz")
-    return qw, qz
 
+    return x, (w, z)
```

Comment:
Ah just saw it here. Then feel free to just remove its definition from inside `confounder_ppca`